### PR TITLE
added basic implementation for Server sent events on the server side.

### LINKED ISSUE #2137: Server Sent Events support
Add support for Server Sent Events on the server side.

### PR DESCRIPTION
**Pull request by devsprint**
Fixes: #2137  - basic implementation for SSE on server side.

### CODE REVIEW

File: zio-http/src/main/scala/zio/http/sse/ServerSentEvent.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      sb.append("retry: ").append(r).append('\n')
+    }
+    sb.append('\n').toString
+  }
+}
+
+object ServerSentEvent {
+  def apply(data: String): ServerSentEvent = new ServerSentEvent(data)
+
+  def heartbeat(): ServerSentEvent = new ServerSentEvent("")
```
Comment (jdegoes) [inline zio-http/src/main/scala/zio/http/sse/ServerSentEvent.scala]:
```suggestion
  def heartbeat: ServerSentEvent = ServerSentEvent("")
```

File: zio-http/src/main/scala/zio/http/sse/ServerSentEvent.scala [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    }
+    retry.foreach { r =>
+      sb.append("retry: ").append(r).append('\n')
+    }
+    sb.append('\n').toString
+  }
+}
+
+object ServerSentEvent {
+  def apply(data: String): ServerSentEvent = new ServerSentEvent(data)
```
Comment (jdegoes) [inline zio-http/src/main/scala/zio/http/sse/ServerSentEvent.scala]: Why not rely on `case class` apply?

File: zio-http/src/main/scala/zio/http/Response.scala [hunk 3]
```diff
@@ -405,7 +408,17 @@ object Response {
...
 
-  private lazy val contentTypeJson: Headers = Headers(Header.ContentType(MediaType.application.json).untyped)
-  private lazy val contentTypeHtml: Headers = Headers(Header.ContentType(MediaType.text.html).untyped)
-  private lazy val contentTypeText: Headers = Headers(Header.ContentType(MediaType.text.plain).untyped)
+  /**
+   * Creates a response with content-type set to text/event-stream
+   * @param data
+   *   \- stream of data to be sent as Server Sent Events
+   */
+  def sse(data: ZStream[Any, Nothing, ServerSentEvent]): Response =
```
Comment (jdegoes) [inline zio-http/src/main/scala/zio/http/Response.scala]: Maybe: `fromServerSentEvents`? (Not sure, if `sse` is commonly understood by those interested in using the protocol, then it's probably fine as `sse`.)

File: zio-http/src/main/scala/zio/http/sse/ServerSentEvent.scala (hunk 1 above)
Comment (devsprint) [reply zio-http/src/main/scala/zio/http/sse/ServerSentEvent.scala]: done

File: zio-http/src/main/scala/zio/http/sse/ServerSentEvent.scala (hunk 2 above)
Comment (devsprint) [reply zio-http/src/main/scala/zio/http/sse/ServerSentEvent.scala]: right

File: zio-http/src/main/scala/zio/http/Response.scala (hunk 3 above)
Comment (devsprint) [reply zio-http/src/main/scala/zio/http/Response.scala]: even I think it is commonly used, `fromServerSentEvents` is much better.

File: zio-http-example/src/main/scala/example/SSEServer.scala [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter.ISO_LOCAL_TIME
+
+import zio.{ExitCode, Schedule, URIO, ZIOAppDefault, durationInt}
+
+import zio.stream.ZStream
+
+import zio.http._
+import zio.http.sse.ServerSentEvent
```
Comment (jdegoes) [inline zio-http-example/src/main/scala/example/SSEServer.scala]: Just one more quick comment here: if there's only ever going to be one thing in `sse`, then the package doesn't really pull its weight, and we should consider moving its sole data type to `zio.http`. What do you think?

Comment (devsprint) [reply zio-http-example/src/main/scala/example/SSEServer.scala]: I was thinking that maybe we have to provide some encoding/decoding support for Server Sent Event, but I guess we can move it for now in `zio.http` and update it later if is need it.

**Review by jdegoes — APPROVED** [review]

**Comment by jdegoes** [comment]
Thanks for your work on this!

**Pull request merged**